src/Store/StreamDoctrineDbalStore.php

```diff
@@ -283,21 +292,38 @@ function () use ($messages): void {
                     $this->executeSave($columns, $placeholders, $parameters, $types, $this->connection);
                 }
 
-                foreach ($achievedUntilPlayhead as $stream => $playhead) {
+                $subselect = null;
+
+                foreach ($achievedUntilEventId as $stream => $eventId) {
+                    if ($subselect === null) {
+                        if ($this->connection->getDatabasePlatform() instanceof MySQLPlatform) {
+                            $subselect = sprintf(
+                                'SELECT t.id FROM (SELECT * FROM %s) AS t WHERE t.event_id = :event_id',
+                                $this->config['table_name'],
+                            );
+                        } else {
+                            $subselect = sprintf(
+                                'SELECT id FROM %s WHERE event_id = :event_id',
```

Comment:
We already had the possibilty to make such a query with `stream` & `playhead` combined. So an extra `eventId` would not be necessary for this to work, right?

Nonetheless i think the eventId could be useful.

Reply:
Yes, it is necessary. 
In the stream store, playhead is optional and can be null. This means you cannot search for this combination.

Reply:
True, forgot again that playhead is not required there